### PR TITLE
[WIP] Verification scripts for post K8S install GPU functionality

### PR DESCRIPTION
**Pull request by yangatgithub**
A script to check GPU plug-in is correctly installed during K8s provisioning. It runs "nvidia-smi" on GPU in the cluster then compare the successful count to the total number of GPUs.

### CODE REVIEW

**Comment by supertetelman** [comment]
We should consider adding this to the k8s documenation.

File: tests/cluster-gpu-test-job.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  name: cluster-gpu-tests
+spec:
+  parallelism: 4 # DYNAMIC_PARALLELISM
+  completions: 4 # DYNAMIC_COMPLETIONS
+  backoffLimit: 5
+  template:
+    spec:
+      containers:
+        - name: tensorflow-container
+          image: nvcr.io/nvidia/cuda:10.0-devel
```
Comment (supertetelman) [inline tests/cluster-gpu-test-job.yml]: Is 10.0 a reasonable firmware version or should we go lower to increase compatability?

File: tests/cluster-gpu-test-job.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+metadata:
+  name: cluster-gpu-tests
+spec:
+  parallelism: 4 # DYNAMIC_PARALLELISM
+  completions: 4 # DYNAMIC_COMPLETIONS
+  backoffLimit: 5
+  template:
+    spec:
+      containers:
+        - name: tensorflow-container
```
Comment (supertetelman) [inline tests/cluster-gpu-test-job.yml]: Change container name

File: scripts/k8s_verify_gpu.sh [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+CLUSTER_VERIFY_NS=cluster-gpu-verify
+
+# Ensure we start in the correct working directory
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+ROOT_DIR="${SCRIPT_DIR}/.."
+cd "${ROOT_DIR}" || exit 1
+TESTS_DIR=$ROOT_DIR/tests
+
+job_name=$(cat $TESTS_DIR/cluster-gpu-test-job.yml | grep -A1 metadata | awk '{print $2}')
+echo "job_name=$job_name"
```
Comment (supertetelman) [inline scripts/k8s_verify_gpu.sh]: Maybe we could pretty print this a bit?

File: scripts/k8s_verify_gpu.sh [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+# loop through all pod from each node
+i=1
+while [ $i -le $total_gpus ]; do
+    kubectl -n ${CLUSTER_VERIFY_NS} logs -f ${string_array[$k]}
+    let i=i+1
+done
+
+kubectl delete ns ${CLUSTER_VERIFY_NS}
+
+echo "Number of Nodes: ${number_gpu_nodes}"
```
Comment (supertetelman) [inline scripts/k8s_verify_gpu.sh]: We should probably export these variables in a way other scripts could use. That'd make it easier for Ansible or some other testing harness to consume these.

File: tests/cluster-gpu-test-job.yml (hunk 2 above)
Comment (yangatgithub) [reply tests/cluster-gpu-test-job.yml]: Done

**Comment by dholt** [comment]
@bulldude test

**Comment by bulldude** [comment]
Got it.

Andrew

________________________________

**Comment by supertetelman** [comment]
Looks like we accidentally pulled some commits into this PR from a non-master branch. We'll need to fix this up before a merge.

**Comment by yangatgithub** [comment]
Open a new PR to track this enhancement: https://github.com/NVIDIA/deepops/pull/420, close this one.

**Pull request closed without merging**